charts/cryostat/tests/deployment_test.yaml

```diff
@@ -0,0 +1,164 @@
+    asserts:
+      - equal:
+          path: spec.template.spec.imagePullSecrets[0].name
+          value: "myregistrykey"
+
+  - it: should configure the OpenShift OAuth Proxy when authentication.openshift is enabled
+    set:
+      authentication.openshift.enabled: true
+    asserts:
+      - matchRegex:
+          path: spec.template.spec.containers[?(@.name=='cryostat-authproxy')].image
+          pattern: "quay.io/cryostat/openshift-oauth-proxy:.*"
+
+  - it: should configure the OAuth2 Proxy when OpenShift OAuth is disabled and OAuth2 is enabled
+    set:
+      authentication.openshift.enabled: false
+      authentication.oauth2.enabled: true
+    asserts:
+    - exists:
+        path: spec.template.spec.containers[?(@.name=='cryostat-authproxy')]
+    - matchRegex:
+        path: spec.template.spec.containers[?(@.name=='cryostat-authproxy')].image
+        pattern: "quay.io/oauth2-proxy/oauth2-proxy:.*"
+
+  - it: should apply Kubernetes specific settings when configured
+    set:
+      nodeSelector:
+        disktype: ssd
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: disktype
+                operator: In
+                values:
+                - ssd
+      tolerations:
+      - key: "key"
+        operator: "Equal"
+        value: "value"
+        effect: "NoSchedule"
+    asserts:
+      - exists:
+          path: spec.template.spec.nodeSelector
+      - exists:
+          path: spec.template.spec.affinity
+      - exists:
+          path: spec.template.spec.tolerations
+
+  - it: should ensure the datasource container has correct environment variables and resources
+    asserts:
+      - equal:
+          path: spec.template.spec.containers[?(@.name=='cryostat-jfr-datasource')].env[?(@.name=='QUARKUS_HTTP_PORT')].value
+          value: "8800"
+      - exists:
+          path: spec.template.spec.containers[?(@.name=='cryostat-jfr-datasource')].resources
+
+  - it: should validate security contexts for each container
+    asserts:
+      - equal:
+          path: spec.template.spec.securityContext.runAsNonRoot
+          value: true
+      - exists:
+          path: spec.template.spec.containers[?(@.name=='cryostat')].securityContext.capabilities.drop
+      - equal:
+          path: spec.template.spec.containers[?(@.name=='cryostat')].securityContext.capabilities.drop[0]
+          value: "ALL"
+
+  - it: should verify image pull policies for each container
+    asserts:
+      - equal:
+          path: spec.template.spec.containers[?(@.name=='cryostat')].imagePullPolicy
+          value: Always
+      - equal:
+          path: spec.template.spec.containers[?(@.name=='cryostat-authproxy')].imagePullPolicy
+          value: Always
+      - equal:
+          path: spec.template.spec.containers[?(@.name=='cryostat-storage')].imagePullPolicy
+          value: Always
```

Comment:
The pull policies get set to `IfNotPresent` for release versions, but `Always` in between for development snapshots. Could that behaviour be accounted for in this test?